src/gz/bufread.rs

```diff
@@ -366,9 +369,10 @@ impl<R: BufRead> Read for GzDecoder<R> {
                                     return Err(err);
                                 }
                             };
+                            headers.push(header);
```

Comment:
Even if performance wouldn't be an issue, these headers contain a few allocations themselves and I can imagine that archives with millions of members would now see a problem with memory consumption where previously they would not have an issue.

In order to prevent regression in that regard, I think this feature must be opt-in, controllable from the `MutliGzDecoder` which already sets the `multi` flag.